src/Contexts/AuthDialogContext/index.jsx

```diff
@@ -0,0 +1,79 @@
+import React, { createContext, useContext, useState } from "react";
+import Login from "../../components/Authentication/Login";
+import SignUp from "../../components/Authentication/SignUp";
+import ModalWrapper from "../../components/ModalWrapper";
+import axios from "axios";
+import utils from '../../utils'
+export const AuthContext = createContext({
+  showModal: false,
+  setShowModal: () => {},
+  authType: null,
+  setAuthType: () => {},
+});
+
+export const AuthProvider = (props) => {
+  const [authType, setAuthType] = useState(null);
+  const [isLoading, setIsLoading] = useState(false);
+  const [error, setError] = useState({ errorExists: false, errorMessage: "" });
+  const [user,setUser] = useState(null);
+
+  const loginHandler = async (user) => {
+    setIsLoading(true);
+    try {
+      const res = await axios.post("/api/auth/login", {
```

Comment:
this could have been in a different function like login service and then you import that function here and provide the form data.